packages/grid/_modules_/grid/components/toolbar/GridFilterToolbarButton.tsx

```diff
@@ -51,9 +51,12 @@ export const GridFilterToolbarButton = React.forwardRef<HTMLButtonElement, Butto
               ...(lookup[item.columnField!] && (
                 <li key={item.id}>
                   {`${lookup[item.columnField!].headerName || item.columnField}
-                  ${apiRef!.current.getLocaleText(
-                    `filterOperator${capitalize(item.operatorValue!)}` as GridTranslationKeys,
-                  )}
+                  ${
```

Comment:
This is better as it avoids modifying the model
```tsx
const getOperatorLabel = (item: GridFilterItem):string=>
        lookup[item.columnField!].filterOperators!.find((operator) => operator.value === item.operatorValue)!.label
        || apiRef!.current!.getLocaleText(
          `filterOperator${capitalize(item.operatorValue!)}` as GridTranslationKeys,
        )!.toString();

      return (
        <div>
          {apiRef!.current.getLocaleText('toolbarFiltersTooltipActive')(counter)}
          <ul className={classes.list}>
            {activeFilters.map((item) => ({
              ...(lookup[item.columnField!] && (
                <li key={item.id}>
                  {`${lookup[item.columnField!].headerName || item.columnField}
                  ${getOperatorLabel(item)}
                  ${item.value}`}
                </li>
              )),
            }))}
          </ul>
        </div>
      );
```

Reply:
+1 for keeping the controllable state as lean as possible (what we can a model, BTW the name is weird).